src/chain/shared_state.rs

```diff
@@ -377,23 +429,48 @@ impl SectionProofChain {
         }
         true
     }
-}
 
-impl Debug for SectionProofChain {
-    fn fmt(&self, formatter: &mut Formatter) -> fmt::Result {
-        write!(
-            formatter,
-            "SectionProofChain(len = {})",
-            self.blocks.len() + 1
-        )
+    pub fn last_public_key(&self) -> &BlsPublicKey {
+        self.blocks
+            .last()
+            .map(|block| &block.key)
+            .unwrap_or(&self.genesis_pk)
+    }
+
+    pub fn last_index(&self) -> usize {
+        self.blocks.len()
```

Comment:
isn't the `last_index` shall be `.len() - 1` ?
I feel `last_index` is bit risky, as when we try to use it, we have to be causious that `itself` is not included, especially in case of empty list.

Reply:
No, since there is always the genesis index. We could use 'self.len() -1' as well.

Reply:
well, this last_index will be used as the end in the later on implemented slice function, which does not include the end.
however, normally, we expect the `index` is including self.
I feel this may cause some mis-understanding ?